pkg/deploy/deploy.go

```diff
@@ -398,6 +405,61 @@ func createJobFromCronjob(k8sClient dynamic.Interface, res *unstructured.Unstruc
 	return jobCreated.GetName(), nil
 }
 
+func withAwaitableJob(timeout time.Duration, apply applyFunction) applyFunction {
+	return func(clients *k8sClients, res resourceutil.Resource, deployConfig utils.DeployConfig) error {
+		gvr, err := resourceutil.FromGVKtoGVR(clients.discovery, res.Object.GroupVersionKind())
+		if err != nil {
+			return err
+		}
+
+		// Register a watcher and start to listen for events for the gvr
+		// if res is an annotated with awaitCompletionAnnotation job
+		var watchEvents <-chan watch.Event
+		startTime := time.Now()
+		_, awaitCompletionFound := res.Object.GetAnnotations()[awaitCompletionAnnotation]
+		if res.GroupVersionKind.Kind == "Job" && awaitCompletionFound {
+			watcher, err := clients.dynamic.Resource(gvr).
+				Namespace(res.Object.GetNamespace()).
+				Watch(context.TODO(), metav1.ListOptions{})
+			if err != nil {
+				return err
+			}
+			watchEvents = watcher.ResultChan()
+			fmt.Println("Registered watcher for job:", res.Object.GetName())
+		}
+
+		if err := apply(clients, res, deployConfig); err != nil {
+			return err
+		}
+
+		if watchEvents == nil {
+			return nil
+		}
+
+		// Consume watcher events and wait for the job to complete or exit because of timeout
+		for {
+			select {
+			case event := <-watchEvents:
+				u := event.Object.(*unstructured.Unstructured)
+				var job batchv1.Job
+				runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &job)
```

Comment:
This function returns an error, it should be handled

Reply:
:+1: